src/main/java/com/shilymily/bms/entity/Company.java

```diff
@@ -0,0 +1,32 @@
+package com.shilymily.bms.entity;
+
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+
+/**
+ * Created by hmaskai on 9/8/18.
+ */
+@Entity
+@Table(name = "companies")
+@Setter @AllArgsConstructor @NoArgsConstructor @ToString
+public class Company {
+    @Id
+    @GeneratedValue(strategy= GenerationType.AUTO)
+    Integer company_id;
```

Comment:
Why is the primary key ID instead of the name? I don't see any advantage using the ID.

Reply:
Its always good to use numbers as ID...take an instance...if two users of this platform have the sam company...they wont be able to...

Reply:
If you are using the companyId as the key then if a user tries to add an already existing company, it will be a successful add and you will have duplicates.
The key should be the companyName.

Reply:
I have already added a unique constraint on the company name.... check